File: app-java/src/test/java/tech/thdev/java_udemy_sample/class_02/_01_AbstractTest.java
package tech.thdev.java_udemy_sample.class_02;

import org.junit.Test;

/**
 * Created by tae-hwan on 09/10/2016.
 */

public class _01_AbstractTest {

    @Test
    public void test() throws Exception {
        AbstractSample abstractSample = new AbstractSample();
        abstractSample.attributeA = "attributeA";
        System.out.println(abstractSample.attributeA);
        System.out.println(abstractSample.attributeB);
        abstractSample.methodA();
        System.out.println(abstractSample.methodB());
    }
}

File: app-kotlin/src/test/java/tech/thdev/kotlin_udemy_sample/class_02/_01_AbstractTest.kt
package tech.thdev.kotlin_udemy_sample.class_02

import org.junit.Test

/**
 * Created by tae-hwan on 09/10/2016.
 */

class _01_AbstractTest {

    @Test
    @Throws(Exception::class)
    fun test() {
        val sample = AbstractSample()

        sample.attributeA = "Test A"
        println(sample.attributeA)
//        sample.attributeB = "Test B"
        println(sample.attributeB)
        sample.methodA()
        println(sample.methodB())
    }
}